libs/cases/frontend/view/src/lib/components/case-browse/case-browse.component.ts

```diff
@@ -1,76 +1,193 @@
+		if (!this.showPriorityEmergency()) {
+			cases = cases.filter((_case) => _case.priority != 'EMERGENCY');
+		}
+		if (!this.showPriorityHigh()) {
+			cases = cases.filter((_case) => _case.priority != 'HIGH');
+		}
+		if (!this.showPriorityMedium()) {
+			cases = cases.filter((_case) => _case.priority != 'MEDIUM');
+		}
+		if (!this.showPriorityLow()) {
+			cases = cases.filter((_case) => _case.priority != 'LOW');
+		}
+
+		if (!this.showTypePlanned()) {
+			cases = cases.filter((_case) => _case.type != 'PLANNED');
+		}
+		if (!this.showTypeIncident()) {
+			cases = cases.filter((_case) => _case.type != 'INCIDENT');
+		}
+		if (!this.showTypeAnnotation()) {
+			cases = cases.filter((_case) => _case.type != 'ANNOTATION');
+		}
+
+		if (!this.showStatusOpen()) {
+			cases = cases.filter((_case) => _case.status != 'OPEN');
+		}
+		if (!this.showStatusInProgress()) {
+			cases = cases.filter((_case) => _case.status != 'INPROGRESS');
+		}
+		if (!this.showStatusOnHold()) {
+			cases = cases.filter((_case) => _case.status != 'ONHOLD');
+		}
+		if (!this.showStatusDone()) {
+			cases = cases.filter((_case) => _case.status != 'DONE');
+		}
+		if (!this.showStatusOverdue()) {
+			cases = cases.filter((_case) => _case.status != 'OVERDUE');
+		}
+		if (!this.showStatusCancelled()) {
+			cases = cases.filter((_case) => _case.status != 'CANCELLED');
+		}
+		if (!this.showStatusArchived()) {
+			cases = cases.filter((_case) => _case.status != 'ARCHIVED');
+		}
+
+		const statusOrder = [
+			'OPEN',
+			'INPROGRESS',
+			'OVERDUE',
+			'ONHOLD',
+			'DONE',
+			'CANCELLED',
+			'ARCHIVED',
+		];
+		const priorityOrder = [ 'EMERGENCY', 'HIGH', 'MEDIUM', 'LOW' ];
+
+		cases.sort((a, b) => {
+			const statusAIndex = statusOrder.indexOf(a.status);
+			const statusBIndex = statusOrder.indexOf(b.status);
+
+			if (statusAIndex === statusBIndex) {
+				const priorityAIndex = priorityOrder.indexOf(a.priority.toUpperCase());
+				const priorityBIndex = priorityOrder.indexOf(b.priority.toUpperCase());
+
+				if (priorityAIndex === priorityBIndex) {
+					return a.id - b.id;
+				} else if (priorityAIndex === -1) {
+					return 1;
+				} else if (priorityBIndex === -1) {
+					return -1;
+				}
+				return priorityAIndex - priorityBIndex;
+			} else if (statusAIndex === -1) {
+				return 1;
+			} else if (statusBIndex === -1) {
+				return -1;
+			}
+			return statusAIndex - statusBIndex;
+		});
+
+		return cases;
+	});
+
+	getStatusClasses(_case: ICaseResponse) {
+		return {
+			emergency: _case.priority === 'EMERGENCY',
+			'status-open': _case.status === 'OPEN',
+			'status-inprogress': _case.status === 'INPROGRESS',
+			'status-overdue': _case.status === 'OVERDUE',
+			'status-onhold': _case.status === 'ONHOLD',
+			'status-done': _case.status === 'DONE',
+			'status-cancelled': _case.status === 'CANCELLED',
+			'status-archived': _case.status === 'ARCHIVED',
+		};
+	}
+
+	flipShowPriorityEmergency() {
+		this.showPriorityEmergency.set(!this.showPriorityEmergency());
```

Comment:
I think the `update` method would be better for this (not sure but there might be race conditions if you press the button really fast). Like this would be better:
```
	this.showPriorityEmergency.update(value => !value);
```